api/models.py

```diff
@@ -229,6 +264,74 @@ def to_dict(self, for_user=None):
 
         return d
 
+class UserSkill(db.Model):
+    __tablename__ = 'user_skills'
+
+    id = db.Column(db.Integer, primary_key=True, autoincrement=True)
+
+    user_id = db.Column(db.Integer, db.ForeignKey(User.id), nullable=False)
+    skill = db.Column(db.String(21))
+
+    added_at = db.Column(db.DateTime, nullable=False, default=datetime.utcnow)
+
+class UserAchievement(db.Model, GeneratedKeyMixin):
+    __tablename__ = 'user_achievements'
+
+    REQUIRED_FIELDS = ['achievement']
+
+    id = db.Column(db.Integer, primary_key=True, autoincrement=True)
+    key = db.Column(db.String(64), nullable=False)
+
+    user_id = db.Column(db.Integer, db.ForeignKey(User.id), nullable=False)
+    from_year = db.Column(db.Integer, nullable=True)
+    from_month = db.Column(db.Integer, nullable=True)
+    to_year = db.Column(db.Integer, nullable=True)
+    to_month = db.Column(db.Integer, nullable=True)
+    achievement = db.Column(db.String(210), nullable=False)
+
+    def to_dict(self):
+        return {
+            'key': self.key,
+            'achievement': self.achievement,
+            'from_year': self.from_year,
+            'from_month': self.from_month,
+            'to_year': self.to_year,
+            'to_month': self.to_month,
+        }
+
+    @classmethod
+    def validate_dict(cls, d):
+        validated = {}
+        for k in ['achievement']:
+            if k not in d:
+                continue
```

Comment:
`achievement` is in `REQUIRED_FIELDS`, but here it seems it's not mandatory. I see that it's "mandated" in the set functions, but I don't know if something can be done here.

Mmmm... maybe you're right and you validate here what you have and enforce "required" from the "set" functions.

Reply:
Yes, here I validate whatever I receive. REQUIRED_FIELDS is what makes sure the fields are there.

Maybe it's not great... I am open to suggestions on how to improve, but then we would change it everywhere, to keep consistency.

Reply:
Nah, I think it's ok. Validate validates the fields, and the "set" fields check if you're setting the required fields.